src/tiled/terrainbrush.cpp

```diff
@@ -371,6 +372,26 @@ void TerrainBrush::updateBrush(QPoint cursorPos, const QVector<QPoint> *list)
         int x = p.x(), y = p.y();
         int i = y*layerWidth + x;
 
+        // to support isometric staggered, make edges into variables
+        QPoint upPoint(p.x(),p.y()-1);
+        QPoint bottomPoint(p.x(),p.y()+1);
+        QPoint leftPoint(p.x()-1,p.y());
+        QPoint rightPoint(p.x()+1,p.y());
+
+        StaggeredRenderer* renderer = dynamic_cast<StaggeredRenderer*>(mapDocument()->renderer());
+        if (renderer) {
+            upPoint = renderer->topRight(p.x(),p.y());
+            bottomPoint = renderer->bottomLeft(p.x(),p.y());
+            leftPoint = renderer->topLeft(p.x(),p.y());
+            rightPoint = renderer->bottomRight(p.x(),p.y());
+        }
+
+        int upperIndex = upPoint.y()*layerWidth + upPoint.x();
+        int bottomIndex = bottomPoint.y()*layerWidth + bottomPoint.x();
+        int leftIndex = leftPoint.y()*layerWidth + leftPoint.x();
+        int rightIndex = rightPoint.y()*layerWidth + rightPoint.x();
+
+
         // if we have already considered this point, skip to the next
         // TODO: we might want to allow re-consideration if prior tiles... but not for now, this would risk infinite loops
         if (checked[i])
```

Comment:
Please move up this check since we don't need to find out the neighbors if we have already checked `i` (you can also inline the calculation of `i` since I don't believe the variable is used anywhere else anymore).